x/feepay/types/msg.go

```diff
@@ -26,6 +26,18 @@ func (msg MsgRegisterFeePayContract) Type() string { return TypeMsgRegisterFeePa
 
 // ValidateBasic runs stateless checks on the message
 func (msg MsgRegisterFeePayContract) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.SenderAddress); err != nil {
+		return err
+	}
+
+	if _, err := sdk.AccAddressFromBech32(msg.Contract.ContractAddress); err != nil {
+		return err
+	}
+
+	if msg.Contract.WalletLimit < 0 || msg.Contract.WalletLimit > 1_000_000 {
```

Comment:
if WalletLimit is a uint then <0 is not possible. -  <1 would be better here (as 0 interactions does not make sense? unless you were thinking about the admin disabling it temporarily).

Reply:
Ah yes, less than zero makes no sense. I'll just ensure it is greater than or equal to 1 and less than a million. Good catch!